src/hub/Hub.sol

```diff
@@ -390,6 +390,13 @@ contract Hub is BatchedMulticall, Auth, Recoverable, IHub, IHubRequestManagerCal
 
         (bool isPositive, uint128 diff) = holdings.update(poolId, scId, assetId);
         _updateAccountingValue(poolId, scId, assetId, isPositive, diff);
+
+        (bool isSnapshot,) = holdings.snapshot(poolId, scId, assetId.centrifugeId());
+
+        if (isSnapshot) {
+            ISnapshotHook hook = holdings.snapshotHook(poolId);
+            if (address(hook) != address(0)) hook.onSync(poolId, scId, assetId.centrifugeId());
+        }
```

Comment:
Shouldn't we call `holding.setSnapshot` as we do in HubHelpers? If not, should not add this hook call inside `holding.update()`?

My point is that we're calling to `hook.onSync()` from two different places in code: holdings and hub, and that's weird.